### PR TITLE
fix: edit profile

### PR DESCRIPTION
**Pull request by sahas-01**
The following has been modified in edit profile:
- usernames can now accept special characters
- profile URL when edited showed the invalid URL error, which has been handled

### CODE REVIEW

File: src/components/UserProfile/Socials.tsx [hunk 1]
```diff
@@ -74,6 +76,16 @@ const Socials: FC<ISocialsProps> = (props) => {
 					);
 				})
 			}
+			{
+				errorCheck ?
+					<Alert
+						className='mt-4 h-[40px] py-2 px-5 text-bodyBlue text-sm rounded-[4px]'
+						description={errorCheck}
+						type='info'
```
Comment (KanishkaRajputd) [inline src/components/UserProfile/Socials.tsx:84]: use message instead of description

File: src/components/UserProfile/BasicInformation.tsx [hunk 2]
```diff
@@ -221,6 +222,16 @@ const BasicInformation: FC<IBasicInformationProps> = (props) => {
 					</article>
 				</div>
 			</div>
+			{
+				errorCheck ?
+					<Alert
+						className='mt-4 h-[40px] py-2 px-5 text-bodyBlue text-sm rounded-[4px]'
```
Comment (KanishkaRajputd) [inline src/components/UserProfile/BasicInformation.tsx:228]: user message instead of description

File: src/ui-components/SocialLinks.tsx [hunk 3]
```diff
@@ -38,17 +38,52 @@ const SocialLink: FC<ISocialLink> = (props) => {
 	const { link: handle, className, type, disable, iconClassName } = props;
 
 	let link = '';
+	let username = '';
+
 	switch (type) {
 	case ESocialType.TWITTER:
-		link = `https://twitter.com/${handle}`;
+		if (handle.startsWith('https://')) {
```
Comment (alphainfinitus) [inline src/ui-components/SocialLinks.tsx:45]: please extract this code to a function

File: src/components/UserProfile/Socials.tsx [hunk 4]
```diff
@@ -13,10 +13,12 @@ interface ISocialsProps {
     loading: boolean;
     setProfile: React.Dispatch<React.SetStateAction<ProfileDetails>>
     profile: ProfileDetails;
+	errorCheck?: string | undefined;
 }
 
 const Socials: FC<ISocialsProps> = (props) => {
-	const { loading, profile, setProfile } = props;
+	const { loading, profile, setProfile , errorCheck } = props;
+	console.log('errorCheck', errorCheck);
```
Comment (alphainfinitus) [inline src/components/UserProfile/Socials.tsx:21]: lingering console

File: src/components/UserProfile/BasicInformation.tsx [hunk 5]
```diff
@@ -221,6 +222,16 @@ const BasicInformation: FC<IBasicInformationProps> = (props) => {
 					</article>
 				</div>
 			</div>
+			{
+				errorCheck ?
```
Comment (alphainfinitus) [inline src/components/UserProfile/BasicInformation.tsx:226]: Please remove unncessary ternary operator
`errorCheck ?` -> `errorCheck &&`

File: src/components/UserProfile/Socials.tsx (hunk 4 above)
Comment (sahas-01) [reply src/components/UserProfile/Socials.tsx:21]: Resolved, pls check

File: src/components/UserProfile/Socials.tsx (hunk 1 above)
Comment (sahas-01) [reply src/components/UserProfile/Socials.tsx:84]: Resolved, pls check

File: src/components/UserProfile/BasicInformation.tsx (hunk 5 above)
Comment (sahas-01) [reply src/components/UserProfile/BasicInformation.tsx:226]: Resolved, pls check

File: src/components/UserProfile/EditProfile.tsx [hunk 6]
```diff
@@ -43,43 +43,53 @@ const EditProfileModal: FC<IEditProfileModalProps> = (props) => {
...
+		socialsError: ''
+	});
 	const userDetailsContext = useUserDetailsContext();
 	const [username, setUsername] = useState<string>(userDetailsContext.username || '');
 	const router = useRouter();
 
 	const validateData = ( image: string | undefined, social_links: ISocial[] | undefined) => {
 
 		// eslint-disable-next-line no-useless-escape
-		const regex = new RegExp(/[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)/gi);
```
Comment (KanishkaRajputd) [inline src/components/UserProfile/EditProfile.tsx:54]: here check with validator.isUrl

Comment (sahas-01) [reply src/components/UserProfile/EditProfile.tsx:54]: Resolved, pls check

**Review by alphainfinitus — APPROVED** [review]

**Pull request merged**